Missionframework/modules/03_persistence/fnc/fn_persistence_removeFromPersistence.sqf

```diff
@@ -0,0 +1,36 @@
+/*
+    KPLIB_fnc_persistence_removeFromPersistence
+
+    File: fn_persistence_removeFromPersistence.sqf
+    Author: KP Liberation Dev Team - https://github.com/KillahPotatoes
+    Date: 2019-03-30
+    Last Update: 2019-03-30
+    License: GNU General Public License v3.0 - https://www.gnu.org/licenses/gpl-3.0.html
+    Public: Yes
+
+    Description:
+        Removes object from persistence system.
+
+    Parameter(s):
+        _object - Object to remove from persistence [OBJECT]
+
+    Returns:
+        Object was removed from persistence [BOOL]
+*/
+params [
+    ["_object", objNull, [objNull]]
+];
+
+if (_object isKindOf "CAManBase") exitWith {
+    private _idx = (KPLIB_persistence_units find _object);
+    if (_idx != -1) then {
+        KPLIB_persistence_units set [_idx, objNull];
+    };
+};
+
+private _idx = (KPLIB_persistence_objects find _object);
+if (_idx != -1) then {
+    KPLIB_persistence_objects set [_idx, objNull];
+};
+
```

Comment:
Same with the `;` at the last line in both ifs

And again as something to consider:
```sqf
private _array = [KPLIB_persistence_objects, KPLIB_persistence_units] select (_object isKindOf "CAManBase");
_array deleteAt (_array find _object);

true
```
or
```sqf
([KPLIB_persistence_objects, KPLIB_persistence_units] select (_object isKindOf "CAManBase")) deleteAt (_array find _object);
``` 
`deleteAt` works fine with the find and it doesn't need that much variables and duplicates of code.
Or is it needed for anything special to have an objNull entry and not just a "deletion"?

Reply:
Either I could set it to null or I would have to add `directCall` so it is scheduler safe. With null approach nulls are deleted at save time by unscheduled save call.